### PR TITLE
[14.0][REF] clean up mix with id and object id fix issue #1000

### PR DESCRIPTION
**Pull request by sebastienbeau**
Depends on
- [x] https://github.com/OCA/search-engine/pull/96

Fix issue #1000 

@simahawk @lmignon 

@thibaultrey ;)

### CODE REVIEW

File: shopinvader/models/shopinvader_variant.py [hunk 1]
```diff
@@ -278,4 +278,19 @@ def get_shop_data(self):
     def _get_shop_data(self):
         """Compute shop data base_jsonify parser."""
         exporter = self.env.ref("shopinvader.ir_exp_shopinvader_variant")
-        return self.jsonify(exporter.get_json_parser(), one=True)
+
+        # Same logic as the search-engine implementation
```
Comment (simahawk) [inline shopinvader/models/shopinvader_variant.py]: this is wrong IMO. It should stay in the parser.
Or, it should default to `id` in the search engine implementation IMO.

**Review by simahawk — CHANGES_REQUESTED** [review]

File: shopinvader/models/shopinvader_variant.py (hunk 1 above)
Comment (sebastienbeau) [reply shopinvader/models/shopinvader_variant.py]: Indeed it's weird to have this hack here.
So the solution is to remove the magic addition of the mapping id in the search engine here : https://github.com/OCA/search-engine/pull/96/files#diff-911701f00b818f1c2893a3a3fd17ccb4b26dfe13a6f7126f6741e1887a7e7691R28
And then in the search engine raise an error if the key "id" is missing, ok for you ?

Comment (simahawk) [reply shopinvader/models/shopinvader_variant.py]: yes

Comment (sebastienbeau) [reply shopinvader/models/shopinvader_variant.py:282]: I have a better solution
I can just define a export line for the product id

Comment (sebastienbeau) [reply shopinvader/models/shopinvader_variant.py]: done !

**Comment by simahawk** [comment]
please check my remark on SE PR

**Comment by sebastienbeau** [comment]
@simahawk @lmignon ready for merge

**Review by PierrickBrun — APPROVED** [review]

**Review by simahawk — APPROVED** [review]

**Pull request merged**